rfd/0159-password-status.md

```diff
@@ -66,6 +68,48 @@ ContactAdmin --> End([End])
 
 For clarity, error cases (such as password mismatch or inability to confirm user identity) were omitted from the above diagram.
 
+We are going to implement the following changes to the WebAuthn flow and APIs:
+
+1. Move the responsibility for verifying the old password to the `ChangePassword` RPC call. This will allow us to decouple the verification from setting the new password in the UI (the old password will not be required when generating the challenge). This verification will be specific to the `ChangePassword` call and will not be a part of the generic `LoginFlow`. The
+2. Add a possibility to request user verification through [`ChallengeExtensions`](https://github.com/gravitational/teleport/blob/3a8dcdc52c47b37091796fadb3ca775c06b7933e/api/proto/teleport/mfa/v1/mfa.proto#L27):
+   ```proto
+   message ChallengeExtensions {
+     // ...
+     string user_verification_requirement = 3; // "required", "discouraged", etc.
+   }
+   ```
+   This will automatically become a parameter of the `CreateAuthenticateChallenge` RPC call through the `ChallengeExtensions` field. To make it possible to specify from the client side, we will accept a corresponding `userVerificationRequirement` parameter to the `/v1/webapi/mfa/authenticatechallenge/password` POST endpoint.
```

Comment:
Is there a reason for keeping `/v1/webapi/mfa/authenticatechallenge/password` at this point? Could we remove it and use `/v1/webapi/mfa/authenticatechallenge` in its place?

Reply:
Good question. I felt like it's less risky to allow explicitly setting the verification requirement parameter just in place where we actually need it. If you think it's not adding unnecessary risk, we can remove this thing altogether. (And on the second thought, if we're worried about allowing this for every type of challenge, we can always only allow it for the password change scope.

Reply:
Good points, thanks for looking at it more closely.

Let me give it a try at answering:

1. IsMFARequired can't be used to bypass a check, just to check whether an MFA check is necessary. This is usually targeted at resources. It's like asking "do I need MFA to SSH to this node?" or similar questions. Not really relevant here.

2. I believe setting AllowReuse for PASSWORD_CHANGE would cause a backend error. Even if it passed, the verifier (in our case the ChangePassword RPC) will say that reuse is not allowed, so it wouldn't pass there either.

3. That I didn't know. I suppose we could still do that in the generic RPC by looking at the scope.

In general I prefer removing the /password endpoint and having less API surface. Chances are we'll either need UV in the generic endpoint or some other flag in the /password endpoint, so they will converge, duplicate, and potentially deviate in annoying ways over time.

Reply:
All right, we will use the generic endpoint. Modified the document appropriately.